irohad/ordering/impl/ordering_service_impl.cpp

```diff
@@ -114,9 +115,8 @@ namespace iroha {
     }
 
     OrderingServiceImpl::~OrderingServiceImpl() {
-      std::lock_guard<std::mutex> lock(m);
       is_finished = true;
-      handle.unsubscribe();
+      std::lock_guard<std::mutex> lock(m);
```

Comment:
Why do you lock mutex after changing variable state?

Reply:
Consider the following case:
1. updateTimer's thread is at
https://github.com/hyperledger/iroha/blob/526d5360547cf879a0e603df8d68eb00823b3c58/irohad/ordering/impl/ordering_service_impl.cpp#L109
2. Dtor's thread is at
https://github.com/hyperledger/iroha/blob/526d5360547cf879a0e603df8d68eb00823b3c58/irohad/ordering/impl/ordering_service_impl.cpp#L118
3. Then the updateTimer comes to the play and skips the if-condition and paused at
https://github.com/hyperledger/iroha/blob/526d5360547cf879a0e603df8d68eb00823b3c58/irohad/ordering/impl/ordering_service_impl.cpp#L112
4. Then thread is switched and dtor proceed to execution flip `is_finished` and perform `queue_` destruction
5. Thread switch to the updateTimer and we get UB here